src/components/faceVerification/standalone/hooks/useGiveUpDialog.js

```diff
@@ -0,0 +1,51 @@
+import React, { useCallback, useContext } from 'react'
+import { isUndefined, negate, pickBy } from 'lodash'
+
+import { fireEvent, FV_GIVEUP } from '../../../../lib/analytics/analytics'
+import AsyncStorage from '../../../../lib/utils/asyncStorage'
+import { useDialog } from '../../../../lib/dialog/useDialog'
+import GiveUpDialog from '../components/GiveUpDialog'
+import { FVFlowContext } from '../context/FVFlowContext'
+import { requestIdle } from '../../../../lib/utils/system'
+import useFVRedirect from './useFVRedirect'
+
+const useGiveUpDialog = (navigation, type) => {
+  const { showDialog } = useDialog()
+  const fvRedirect = useFVRedirect()
+  const { isFVFlow } = useContext(FVFlowContext)
+  const { navigate } = navigation
+
+  const onReasonChosen = useCallback(
+    (reason = undefined) => {
+      const data = pickBy({ reason }, negate(isUndefined))
+
+      if (reason && reason !== 'closed') {
+        AsyncStorage.removeItem('hasStartedFV')
+      }
+
+      fireEvent(FV_GIVEUP, { data, surveyType: type })
```

Comment:
if reason is 'closed' ie user closed the modal then the event will still fire, is this intended?
if reason is undefined then the event will still fire.
why would reason be undefined?

Reply:
reason undefined is basically only possible when user closes the modal

mm, not sure if the event should be fired.
@drptiago do we want to have the data for the users who did not complete the survey?